deepmd/pt/loss/denoise.py

```
@@ -8,102 +9,270 @@
+        label["clean_coord"] = input_dict["coord"].clone().detach()
+        label["clean_box"] = input_dict["box"].clone().detach()
+        origin_frac_coord = phys2inter(
+            label["clean_coord"], label["clean_box"].reshape(nbz, 3, 3)
+        )
+        label["clean_frac_coord"] = origin_frac_coord.clone().detach()
+        if self.mask_cell:
+            strain_components_all = torch.zeros(
+                (nbz, 3), dtype=env.GLOBAL_PT_FLOAT_PRECISION, device=env.DEVICE
+            )
+            for ii in range(nbz):
+                cell_perturb_matrix, strain_components = get_cell_perturb_matrix_HEA(
```

Ruff (0.8.2)

Undefined name `get_cell_perturb_matrix_HEA`

(F821)